src/library.js

```diff
@@ -644,9 +644,46 @@ LibraryManager.library = {
 
   system__deps: ['__setErrNo'],
   system: function(command) {
+#if ENVIRONMENT_MAY_BE_NODE
+    if (ENVIRONMENT_IS_NODE) {
+      if (!command) return 1; // shell is available
+
+      var cmdstr = UTF8ToString(command)
+      if (!cmdstr.length) return 0; // this is what glibc seems to do (shell works test?)
+
+      var cp = require('child_process');
+      var ret = cp.spawnSync(cmdstr, [], {shell:true, stdio:'inherit'});
+
+      function _W_EXITCODE(ret, sig) {
+        return ((ret) << 8 | (sig));
```

Comment:
what is this based on? reading the manpage for system I couldn't find a reason for it. Is it just for the test?

Reply:
Both musl and glibc actually return a exit code that is bitpacked `signal:coredump:exitcode` I was kinda surprised as well since the man page doesn't say anything about it.

Reply:
I think node might internally have the raw exit code, but doesn't expose it.

Reply:
Interesting, I didn't know this. Sounds good to return the same exit codes other programs do.

Reply:
Correction to above, this is standard POSIX behaviour. The exit code gets bitpacked by linux during wait system call. Which system() uses internally, the exit code behaviour is documented by the `system() shall return the termination status of the command language interpreter in the format specified by waitpid().` bit.